leptonai/cloudrun/remote.py

```diff
@@ -205,42 +250,69 @@ def start_up(self):
             raise RuntimeError(
                 f"Failed to get openapi for photon {self.photon_id} in time."
             )
-
-        global _refcount
-        _refcount[self.deployment_id] += 1
         logger.debug(f"Remote: photon {self.deployment_id} is ready")
 
-    def alive(self) -> bool:
+    def restart(self) -> None:
         """
-        Returns True if the function is alive, False otherwise.
+        Restarts the current run if it has been closed.
         """
-        if self.client is None:
-            return False
-        else:
-            return self.client.healthz()
+        # Note: there is one caveat in the current implementation, in the sense
+        # that, if someone runs a clean() operation between the launch of this
+        # Remote object and the restart, then the underlying photon will have
+        # been removed, and we will have to recreate the photon. Right now,
+        # we will not handle this case, and we will simply raise an error.
+        if self.healthz():
+            # If the service is still up - don't do anything, just return.
+            return
+        try:
+            if not self.deployment_id:
+                # Never started up before, start up.
+                self._start_up()
+            else:
+                # Resume by re-running the photon
+                self._run_photon(resume=True)
+                if not self.client:
```

Comment:
I think here still needs to call `_create_client` to make sure the photon has finished the `init` stage?

Reply:
Actually maybe it's cleaner to always call `self._create_client()` inside `self._run_photon`

Reply:
True. Let me refactor.